### PR TITLE
feat: Add domain and repository for users, roles and templates with permissions

### PR DESCRIPTION
**Pull request by FirestarJes**

### CODE REVIEW

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public User()
+    {
+        Id = Guid.Empty;
+        ExternalId = Guid.Empty;
+        Name = string.Empty;
+        RoleAssignments = new List<UserRoleAssignment>();
+    }
+
+    public Guid Id { get; set; }
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs:29]: Create UserId and ExternalUserId objects.

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    public Guid ExternalId { get; set; }
+    public string Name { get; set; }
```
Comment (defectiveAi) [inline source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs:31]: Avoid "bag of properties" domain models: remove setters that do not make sense (everything except name), and make ctor that restores an object from DB.

**Review by defectiveAi — APPROVED** [review]

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs (hunk 2 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs:31]: Will fix

Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs]: Fixed

File: source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs (hunk 1 above)
Comment (FirestarJes) [reply source/Energinet.DataHub.MarketParticipant.Domain/Model/Users/User.cs]: Fixed

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**